samples/bookinfo/src/build-services.sh

```diff
@@ -16,88 +16,24 @@
 
 set -ox errexit
 
-display_usage() {
-    echo
-    echo "USAGE: ./build-services.sh <version> <prefix> [-h|--help]"
-    echo "    -h|--help: Prints usage information"
-    echo "    version:   Version of the sample app images (Required)"
-    echo "    prefix:    Use the value as the prefix for image names (Required)"
-}
-
-if [ "$#" -ne 2 ]; then
-  if [ "$1" == "-h" ] || [ "$1" == "--help" ]; then
-    display_usage
-    exit 0
-  else
-    echo "Incorrect parameters" "$@"
-    display_usage
-    exit 1
-  fi
-fi
-
-VERSION=$1
-PREFIX=$2
+# Get to the root directory of the repo...
 SCRIPTDIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+cd $SCRIPTDIR/../../..
 
-# Docker build variables
-ENABLE_MULTIARCH_IMAGES=${ENABLE_MULTIARCH_IMAGES:-"false"}
+h="${BOOKINFO_HUB:?BOOKINFO_HUB must be set}"
+t="${BOOKINFO_TAG:?BOOKINFO_TAG must be set}"
 
-if [ "${ENABLE_MULTIARCH_IMAGES}" == "true" ]; then
-  PLATFORMS="linux/arm64,linux/amd64"	
-  DOCKER_BUILD_ARGS="docker buildx build --platform ${PLATFORMS} --push"
-  # Install QEMU emulators
-  docker run --rm --privileged tonistiigi/binfmt --install all
-  docker buildx rm multi-builder || :
-  docker buildx create --use --name multi-builder --platform ${PLATFORMS}
-  docker buildx use multi-builder
-else
-  DOCKER_BUILD_ARGS="docker build"	
+if [[ ("${h}" == "istio" || "${h}" == "docker.io/istio") && -z "$CI" ]]; then
+  echo "Can only push to prod registry in CI"
+  exit 1
 fi
 
-pushd "$SCRIPTDIR/productpage"
-  ${DOCKER_BUILD_ARGS} --pull -t "${PREFIX}/examples-bookinfo-productpage-v1:${VERSION}" -t "${PREFIX}/examples-bookinfo-productpage-v1:latest" .
-  # flooding
-  ${DOCKER_BUILD_ARGS} --pull -t "${PREFIX}/examples-bookinfo-productpage-v-flooding:${VERSION}" -t "${PREFIX}/examples-bookinfo-productpage-v-flooding:latest" --build-arg flood_factor=100 .
-popd
-
-pushd "$SCRIPTDIR/details"
-  # plain build -- no calling external book service to fetch topics
-  ${DOCKER_BUILD_ARGS} --pull -t "${PREFIX}/examples-bookinfo-details-v1:${VERSION}" -t "${PREFIX}/examples-bookinfo-details-v1:latest" --build-arg service_version=v1 .
-  # with calling external book service to fetch topic for the book
-  ${DOCKER_BUILD_ARGS} --pull -t "${PREFIX}/examples-bookinfo-details-v2:${VERSION}" -t "${PREFIX}/examples-bookinfo-details-v2:latest" --build-arg service_version=v2 \
-	 --build-arg enable_external_book_service=true .
-popd
+# Pass input args to the command. This allows using --push, --load, etc
+env -i TAG="${BOOKINFO_TAG}" HUB="${BOOKINFO_HUB}" \
+  docker buildx bake -f samples/bookinfo/src/docker-bake.hcl $@
 
+if [[ "${BOOKINFO_UPDATE}" == "true" ]]; then
+# Update image references in the yaml files
+  find ./samples/bookinfo/platform -name "*bookinfo*.yaml" -exec sed -i.bak "s#image:.*\\(\\/examples-bookinfo-.*\\):.*#image: ${h//\//\\/}\\1:$t#g" {} +
```

Comment:
nit: We had this before but seems this will leave us with some extra .bak files. Likely a change to run on a Mac, I think the `sed -i` is a Mac thing, that we never cleaned up.